File: Data Structures & Algorithms/binary-search/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    BSR(arr, low, high, target) {
        if(low > high) return -1;
        let mid = low + Math.floor((high-low)/2);
        if(arr[mid] === target) return mid;
        if(arr[low] === target) return low;
        if(arr[high] === target) return high;
        else if(arr[mid] > target) return this.BSR(arr, low, mid-1,target);
        else if(arr[mid] < target) return this.BSR(arr, mid+1, high,target);
    }
    search(arr, target) {
        return this.BSR(arr, 0, arr.length-1, target)
    }
}

File: Data Structures & Algorithms/search-insert-position/submission-1.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        n = len(nums)
        l ,r, = 0, n-1
        while l<=r:
            if l == r:
                if nums[l] == target:
                    return l
                elif nums[l] < target:
                    return l+1
                else:
                    return l-1
            mid = math.floor((l+r)/2)
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                r = mid-1
            else:
                l = mid+1